src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerArgumentCollection.java

```diff
@@ -29,11 +30,21 @@
     @ArgumentCollection
     public AssemblyRegionTrimmerArgumentCollection assemblyRegionTrimmerArgs = new AssemblyRegionTrimmerArgumentCollection();
 
-    protected boolean useMutectAssemblerArgumentCollection() { return false; }
+    public ReadThreadingAssembler createReadThreadingAssembler() {
+        final ReadThreadingAssemblerArgumentCollection rtaac = assemblerArgs;
+        final ReadThreadingAssembler assemblyEngine = rtaac.makeReadThreadingAssembler();
+        assemblyEngine.setDebug(debug);
+        assemblyEngine.setMinBaseQualityToUseInAssembly(minBaseQualityScore);
+
+        return assemblyEngine;
+    }
+
+    protected ReadThreadingAssemblerArgumentCollection getReadThreadingAssemblerArgumentCollection() {
+        return new HaplotypeCallerReadThreadingAssemblerArgumentCollection();
```

Comment:
I would leave this `abstract` and implement it in both `HaplotypeCallerArgumentCollection` and `Mutect2ArgumentCollection`.